graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -240,16 +240,40 @@ def rename_schema(
     schema = build_ast_schema(schema_ast)
     query_type = get_query_type_name(schema)
 
+    # If we suppress a type, any interfaces it implements must be made unqueryable as described in
+    # the module-level comment.
+    # Unfortunately, this must be done outside a schema visitor object because the interfaces field
+    # of ObjectTypeDefinitionNodes and InterfaceTypeDefinitionNodes is a list of NameNodes, which
+    # just indicates the name of the interface rather than give the node itself. This makes it
+    # impossible to find all ancestors when only given such a type definition node. Instead, it's
+    # necessary to use the definitions field of schema_ast to traverse the inheritance hierarchy and
+    # find which interfaces need to be made unqueryable.
+    # See discussion here: https://github.com/graphql-python/graphql-core/issues/124
+    interfaces_to_make_unqueryable = set()
+    interface_and_object_type_name_to_definition_node_map = {
+        node.name.value: node
+        for node in schema_ast.definitions
+        if isinstance(node, (ObjectTypeDefinitionNode, InterfaceTypeDefinitionNode))
+    }
+    for type_name in type_renamings:
+        type_not_suppressed = type_renamings[type_name] is not None
+        if type_not_suppressed:
+            continue
+        if type_name in interface_and_object_type_name_to_definition_node_map:
```

Comment:
This is a bit unfortunate-- so, as it is, we need to compute which interfaces should be made unqueryable _before_ doing any sort of renaming. However, all the error handling for unused type renamings happens _after_ all the renaming is done. If `type_name`, an entry in `type_renamings`, is not in the map, then `type_name` doesn't actually correspond to a type in the schema, and it should have no effect. Now that I look at this again, there might be a better way to organize this but I'm not quite sure how might work best-- open to suggestions! 

(I suppose this also affects the first part, where we might do this in a single conditional.)

Reply:
When `type_name not in interface_and_object_type_name_to_definition_node_map`, the user is attempting to rename something that is not in the schema, right? It seems like that should be an error to me because renaming something that doesn't exist does not make sense. Am I understanding the meaning correctly, and, if so, any reasons this shouldn't be an error?

Reply:
It does end up as an error, but the error handling happens later on. The structure of the code, before this PR, is along the lines of:

1. Go through the schema, applying renamings via the visitor pattern
2. If there are any unused renamings, then raise an error saying which renamings were unused.

Now, we're in a bit of a catch-22 situation: the point of the visitor is that we don't have access to the entire schema-- only individual nodes at any one time. However, we [can't find out which interface nodes to be made unqueryable without](https://github.com/graphql-python/graphql-core/issues/124) `schema_ast.definitions`, which needs to be done outside a visitor object (and therefore before any detection of unused renamings).

The workaround that this PR currently uses is basically ignoring unused type renamings because they'll be handled later on in the helper functions that use visitor objects. Hacky, but I'm not sure if we have other options here.

Reply:
I'm assuming you want to hold off on raising an error message until later because it will be a more complete error message with all the issues in the schema renaming, right? This could definitely benefit from a comment explaining that and I still think it can just be 1 conditional.